api/transaction_handler/metamorph.go

```diff
@@ -70,7 +72,7 @@ func (m *Metamorph) GetTransactionStatus(ctx context.Context, txID string) (stat
 	tx, err = m.Client.GetTransactionStatus(ctx, &metamorph_api.TransactionStatusRequest{
 		Txid: txID,
 	})
-	if err != nil {
+	if err != nil && !strings.Contains(err.Error(), metamorph.ErrNotFound.Error()) {
```

Comment:
IMO it is better to compare using the `errors.Is()` function.

```
if err != nil && !errors.Is(err, metamorph.ErrNotFound.Error) { ... }
```

Reply:
The error isn't `key could not be found` the error is `code = Unknown desc = key could not be found` so I still need `contains` or I need to define that error as `code = Unknown desc = key could not be found`